lib/rules/indent.js

```diff
@@ -807,6 +889,71 @@ module.exports = {
             });
         }
 
+        /**
+        * Ignore first token of each line within an unknown node if its
+        * offset does not depend on another token's offset. If the first token
+        * of a line's offset depends on the offset of another token, make the
+        * last dependent offset depend on the first token of its line.
+        * @param {ASTNode} node Node
+        * @returns {void}
+        */
+        function ignoreUnknownNode(node) {
+            const startLine = node.loc.start.line;
+            const endLine = node.loc.end.line;
+
+            for (let lineNumber = startLine; lineNumber <= endLine; lineNumber++) {
+                const firstTokenOfLine = tokenInfo.firstTokensByLineNumber.get(lineNumber);
+
+                if (!firstTokenOfLine) {
+                    continue;
+                }
+
+                const tokenInNodeRange = (node.range[0] <= firstTokenOfLine.range[0]) &&
+                    (node.range[1] >= firstTokenOfLine.range[1]);
+
+                if (!tokenInNodeRange) {
+                    continue;
+                }
+
+                const dependency = offsets.getFirstDependency(firstTokenOfLine);
+
+                if (dependency) {
+                    let currDependency = dependency,
+                        lastDependency = null;
+
+                    while (
+                        (currDependency = offsets.getFirstDependency(currDependency)) &&
+                        currDependency.loc.start.line > startLine
```

Comment:
The only example I can find where this is required is an abstract class method. For example:
```ts
1 abstract class Foo {
2   public bar() {
3     const baz = 4;
4   }
5 }
```

The first token of line 3, `const` depends on offset of token `{`, found at the end of line 2. The `{` token depends on the offset of token `(`, and this is the last token offset dependency.

In most cases, function/arrow expressions or if statements, the last token offset dependency will be the first token of the line. In the above case this is not true, the last token offset dependency is in the middle of line 2.

So this means when we get the desired offset for line number 3, by calling `getDesiredIndent`, we get the offset of line 2 as if it were not ignored. Line 2 is supposed to have no indention or an offset of 0. This is because we only ignore the first token of every line.

There are three solutions, I can think of, to this problem:
1) We add another token dependency, which is the first token of the line. This is what line 934 does by calling `offsets.matchIndentOf()`. It takes the last dependency and makes it depend on the offset of the first token of the dependent line.
2) We could ignore the last token used when calculating the line's offset. Currently `ignoreToken()` can only ignore the first token of the line, since it calculates the offset based on this assumption.
3) We set the offset of the last token offset dependency to that of its line's current offset, which would be the same as ignoring it.

All of these solutions require finding the last token offset that the current line depends on.

Sorry for the long winded explanation but it helps me ensure I'm using the correct logic. We could even add another if statement that says if the last token dependency is the first token of the line do not call `offsets.matchIndentOf`, since it would be redundant.